Source/Tools/FEXLoader/LinuxSyscalls/FileManagement.cpp

```diff
@@ -59,48 +60,69 @@ namespace JSON {
 }
 
 namespace FEX::HLE {
-  struct open_how;
-
-static bool LoadFile(fextl::vector<char> &Data, const fextl::string &Filename) {
-  int fd = open(Filename.c_str(), O_RDONLY | O_CLOEXEC);
-  if (fd == -1) {
-    return false;
-  }
-
-  struct stat buf;
-  if (fstat(fd, &buf) != 0) {
-    LogMan::Msg::DFmt("Couldn't load configuration file: fstat");
-    close(fd);
-    return false;
-  }
+bool FileManager::RootFSPathExists(const char* Filepath) {
+  LOGMAN_THROW_A_FMT(Filepath && Filepath[0] == '/', "Filepath needs to be absolute");
+  return FHU::Filesystem::ExistsAt(RootFSFD, Filepath + 1);
+}
 
-  auto FileSize = buf.st_size;
+void FileManager::LoadThunkDatabase(fextl::unordered_map<fextl::string, ThunkDBObject>& ThunkDB, bool Global) {
+  auto ThunkDBPath = FEXCore::Config::GetConfigDirectory(Global) + "ThunksDB.json";
+  fextl::vector<char> FileData;
+  if (FEXCore::FileLoading::LoadFile(FileData, ThunkDBPath)) {
+    FileData.push_back(0);
 
-  if (FileSize <= 0) {
-    LogMan::Msg::DFmt("FileSize less than or equal to zero specified");
-    close(fd);
-    return false;
-  }
+    // If the thunksDB file exists then we need to check if the rootfs supports multi-arch or not.
+    const bool RootFSIsMultiarch = RootFSPathExists("/usr/lib/x86_64-linux-gnu/") ||
+      RootFSPathExists("/usr/lib/i386-linux-gnu/");
+
+    fextl::vector<fextl::string> PathPrefixes{};
+    if (RootFSIsMultiarch) {
+      // Multi-arch debian distros have a fairly complex arrangement of filepaths.
+      // These fractal out to the combination of library prefixes with arch suffixes.
+      constexpr static std::array<std::string_view, 4> LibPrefixes = {
+        "/usr/lib",
+        "/usr/local/lib",
+        "/lib",
+        "/usr/lib/pressure-vessel/overrides/lib",
+      };
 
-  Data.resize(FileSize);
-  const auto ReadSize = pread(fd, Data.data(), FileSize, 0);
+      // Arch-Prefixes
+      constexpr static std::array<std::string_view, 2> MultiArchPrefixes = {
+        "i386-linux-gnu", // 32-bit
+        "x86_64-linux-gnu", // 64-bit
+      };
 
-  close(fd);
-  return ReadSize == FileSize;
-}
+      // Now just generate the combination of the two.
+      // We only need to generate 32-bit or 64-bit depending on the operating mode.
+      const auto ArchPrefix = MultiArchPrefixes[Is64BitMode()];
```

Comment:
Why not just `const auto ArchPrefix = Is64BitMode() ? "x86_64..." : "i386..."` ?

Reply:
Dropped the array indexing and just used a ternary.